[DOCS]  Document Batches and Partitioners Functionality for GX Cloud

kwcanuck opened the pull request:
[DSB-475](https://greatexpectations.atlassian.net/browse/DSB-475) requests adding content to the existing GX Cloud documentation that documents how to add partitioners to Data Assets. This PR implements this request. 

## Definition of done
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Appropriate tests and docs have been updated

[DSB-475]: https://greatexpectations.atlassian.net/browse/DSB-475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kwcanuck committed:
Preliminary edits.

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=9052&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/9052/327872ce/391010749bdd4f80ce49795cddc876e3a50ede75.svg)](https://app.codesee.io/r/reviews?pr=9052&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Updated content to match updated UI.

jcampbell reviewed (changes requested):
I'd like to request a name change away from "splitters".

Before we lock that in, I need to confirm with the Name Tsar (Hernan)

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Changed splitter and splitters to partition or partitions

Rachel-Reverie commented on docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:
```diff
@@ -47,27 +47,33 @@ Create a Data Asset to define the data you want GX Cloud to access. Currently, t
+    - Clear **Create temp table** if you don't want to create a temporary database table. Temporary database tables store data temporarily and can improve performance by making queries run faster.
 
-6. Optional. Clear **Test connection** if you don't want to test the Data Asset connection. Testing the connection to the Data Asset is a preventative measure that makes sure the connection configuration is correct. This verification can help you avoid errors and can reduce troubleshooting downtime.
+    - Select **Test connection** if you don't want to test the Data Asset connection. Testing the connection to the Data Asset is a preventative measure that makes sure the connection configuration is correct. This verification can help you avoid errors and can reduce troubleshooting downtime.
```
```suggestion
    - Select **Test connection** if you want to test the Data Asset connection. Testing the connection to the Data Asset is a preventative measure that makes sure the connection configuration is correct. This verification can help you avoid errors and can reduce troubleshooting downtime.
```
Shouldn't they be selecting **Test connection** if they do want to test the Data Asset connection, or leave it unselected it if they do not?

Rachel-Reverie commented on docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:
```diff
@@ -47,27 +47,33 @@ Create a Data Asset to define the data you want GX Cloud to access. Currently, t
-
     - **Table name**: When **Table Asset** is selected, enter a name for the table you're creating in the Data Asset.
+    
+    - **Data Asset name**: Enter a name for the Data Asset. Data Asset names must be unique. If you use the same name for multiple Data Assets, each Data Asset must be associated with a unique Data Source.
```
```suggestion
    - **Data Asset name**: Enter a name for the Data Asset. If you use the same name for multiple Data Assets, each Data Asset must be associated with a unique Data Source.
```
The sentence about Data Asset names needing to be unique is contradicted by the sentence immediately following it.  I think we can cut it out, as the second sentence conveys the point more accurately.

Rachel-Reverie commented on docs/docusaurus/docs/cloud/data_assets/manage_data_assets.md:
```diff
@@ -47,27 +47,33 @@ Create a Data Asset to define the data you want GX Cloud to access. Currently, t
     - **Query**: When **Query Asset** is selected, enter the query that you want to run on the table. 
 
-9. Optional. Select **Add another Data Asset** to add additional tables or queries and repeat step 7.
+9. Select the **Complete Asset** tab to create Expectations and run validations on the entire Data Asset, or select the **Batches** tab to create Expectations and run validations on subsets of Data Asset records and then complete the following fields:
```
I think this should be reworded.  They aren't creating Expectations or running validations from either of these tabs, they're determining how the records in the Data Asset will be organized.
```suggestion
9. Select the **Complete Asset** tab to provide all Data Asset records to your Expectations and validations, or select the **Batches** tab to use subsets of Data Asset records for your Expectations and validations. If you selected the **Batches** tab, complete the following fields:
```

Rachel-Reverie commented on docs/docusaurus/docs/cloud/quickstarts/snowflake_quickstart.md:
```diff
@@ -86,23 +88,31 @@ Create a Data Asset to define the data you want GX Cloud to access within Snowfl
+    - Clear **Create temp table** if you don't want to create a temporary database table. Temporary database tables store data temporarily and can improve performance by making queries run faster.
 
-5. Optional. Clear **Test connection** if you don't want to test the Data Asset connection.
+    - Select **Test connection** if you don't want to test the Data Asset connection. Testing the connection to the Data Asset is a preventative measure that makes sure the connection configuration is correct. This verification can help you avoid errors and can reduce troubleshooting downtime.
```
```suggestion
    - Select **Test connection** if you want to test the Data Asset connection. Testing the connection to the Data Asset is a preventative measure that makes sure the connection configuration is correct. This verification can help you avoid errors and can reduce troubleshooting downtime.
```

Rachel-Reverie commented on docs/docusaurus/docs/cloud/quickstarts/snowflake_quickstart.md:
```diff
@@ -86,23 +88,31 @@ Create a Data Asset to define the data you want GX Cloud to access within Snowfl
     - **Query**: When **Query Asset** is selected, enter the query that you want to run on the table. 
 
-7. Optional. Select **Add table/query** to add additional tables or queries and repeat step 6.
+9. Select the **Complete Asset** tab to create Expectations and run validations on the entire Data Asset, or select the **Batches** tab to create Expectations and run validations on subsets of Data Asset records and then complete the following fields:
```
If you reworded this in the manage_data_assets guide, you should reword it here as well.

kwcanuck replied:
@Rachel-Reverie I added the edit as recommended. Thanks!

kwcanuck committed:
Merge branch 'develop' into splitters

Rachel-Reverie reviewed (approved):
Looks good other than a few minor suggestions.  The only ones that I think require a change are the two places where "don't" appears to have been incorrectly left in a statement; take or leave the others.  Marking this as approved so that you aren't waiting on me after you've made updates.

kwcanuck commented on docs/docusaurus/docs/cloud/quickstarts/snowflake_quickstart.md:
```diff
@@ -86,23 +88,31 @@ Create a Data Asset to define the data you want GX Cloud to access within Snowfl
     - **Query**: When **Query Asset** is selected, enter the query that you want to run on the table. 
 
-7. Optional. Select **Add table/query** to add additional tables or queries and repeat step 6.
+9. Select the **Complete Asset** tab to create Expectations and run validations on the entire Data Asset, or select the **Batches** tab to create Expectations and run validations on subsets of Data Asset records and then complete the following fields:
```
```suggestion
9. Select the **Complete Asset** tab to provide all Data Asset records to your Expectations and validations, or select the **Batches** tab to use subsets of Data Asset records for your Expectations and validations. If you selected the **Batches** tab, complete the following fields:
```

kwcanuck committed:
Apply suggestions from code review

Co-authored-by: Rachel-Reverie <94694058+Rachel-Reverie@users.noreply.github.com>

kwcanuck commented:
@jcampbell I removed instances of _splitter_ and _splitters_ and replaced them with _partition_ or _partitions_ where applicable. Please review the changes, provide feedback, or unblock the PR when time permits.

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Minor edit.

jcampbell reviewed (approved):
Thank you, @kwcanuck . I think we may want to make some additional in-app copy changes ("Split Data by") ... do we have a good way to ensure that this doc will be kept in sync?

kwcanuck commented:
@jcampbell I perform a final test on procedures before they are released to ensure they are accurate and reflect the existing UI. I also monitor Slack channels for announcements of changes and updates.

kwcanuck committed:
Minor edit.

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Updates due to changed UI.

kwcanuck committed:
Merge branch 'develop' into splitters

kwcanuck committed:
Minor edits.

kwcanuck committed:
Merge branch 'splitters' of https://github.com/great-expectations/great_expectations into splitters

kwcanuck committed:
Minor edits.

kwcanuck committed:
Minor edits.
